src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,258 @@
+    // this is essential because we don't want to lose track of file's size for renamed files.
+    _fs.rename = function(oldPath, newPath, callback){
+        _innerRename.call(_fs, oldPath, newPath, function(err){
+            // only update cache once original was successful
+            if(!err){
+                _cache[newPath] = _cache[oldPath];
+                delete _cache[oldPath];
+            }
+            
+            // call callback regardless
+            callback(err);
+        });
+    };
+
+    // returns true/false depending on whether root/index.html exists
+    ProjectStats.hasIndexFile = function() {
+        // Loop through all paths, return true if one of them is root/index.html
+        return !!(Object.keys(_cache)).find(function(path) {
+            return path === _Path.join(_root, "index.html");
+        });
+    };
+
+    // returns an array of filenames for specified extension
+    // adds period if not provided
+    ProjectStats.usesExtension = function(fileExtensions, callback) {
+        var rc = [];
+
+        if(_root){ 
+            //walk the root
+            Object.keys(_cache).filter(function(path){
+                fileExtensions.forEach(function(ext, index){
+                    // check if mime is passed          
+                    if(ext.includes('/')){
+                        // remove the mime and replace it with corresponding extensions
+                        fileExtensions.pop(index, 1);
+                        Array.prototype.push.apply(fileExtensions, _extFromMime(ext));
+                    } else {
+                        if(path.endsWith(ext)){
+                            // check to if this extension has been encountered before
+                            if(rc[ext]){
+                                rc[ext].push(path);
+                            } else {
+                                rc[ext] = Array.from([path]);
+                            }
+                        }
+                    }
+                    
+                });
+            });
+        }
+
+        // manually update the length, since we are using string as keys
+        rc.length = Object.keys(rc).length;
+        if(!rc.length){
+            return callback("No files found!");
+        }
+
+        callback(null,rc);
+    };
+
+    // returns bytes of files in the root   
+    ProjectStats.getTotalProjectSize = function() {
+        // Sum up all sizes in the cache and return
+        return _root ? (Object.values(_cache)).reduce(function(acc, val) {
+            return acc + (val ? val : 0);
+        }) : 0;
+    };
+
+    // returns project file count
+    ProjectStats.getFileCount = function (){
+        return _root ? Object.keys(_cache).length : 0;
+    };
+
+    // function converts mimes into corresponding file formats.
+    // mime format e.g. "text/html". Also supports "text/*", which returs an array of all corresponding types.
+    function _extFromMime(mime) {
```

Comment:
You asked about whether to include the `content` module and for sure you should vs. repeating this code.  It makes it harder to maintain, because we end up with 2 versions of the same thing.  Just include the module at the top in your `define` call.  If you're not sure what I mean, let me know.